test/test_dynamic_shapes.py

```diff
@@ -151,18 +151,27 @@ def create_symbolic_tensor(name, arg, shape_env, source=None, dynamic_dims=None)
 
 def create_symtype(cls, pytype, shape_env, val, duck=True):
     from torch._dynamo.source import ConstantSource
+    source = ConstantSource(f"__testing_only{len(shape_env.var_to_val)}")
     symbol = shape_env.create_symbol(
         val,
-        source=ConstantSource(f"__testing_only{len(shape_env.var_to_val)}"),
+        source=source,
         dynamic_dim=DimDynamic.DUCK if duck else DimDynamic.DYNAMIC,
         constraint_dim=None,
     )
-    return cls(SymNode(
-        symbol,
-        shape_env,
-        pytype,
-        hint=val,
-    ))
+    if cls is SymInt:
```

Comment:
Why does this change?

Reply:
Need to hit the SymInt specific ShapeEnv logic which actually allocates an FX node for it